rubix-core/src/main/java/com/qubole/rubix/core/CachingFileSystem.java

```diff
@@ -115,6 +118,33 @@ public void initialize(URI uri, Configuration conf) throws IOException
     fs.initialize(originalUri, conf);
   }
 
+  public synchronized void initializeClusterManager(Configuration conf, ClusterType clusterType)
+      throws ClusterManagerInitilizationException
+  {
+    if (clusterManager != null) {
+      return;
+    }
+
+    String clusterManagerClassName = CacheConfig.getClusterManagerClass(conf, clusterType);
+    log.info("Initializing cluster manager : " + clusterManagerClassName);
+
+    try {
+      Class clusterManagerClass = conf.getClassByName(clusterManagerClassName);
+      Constructor constructor = clusterManagerClass.getConstructor();
+      ClusterManager clusterManager = (ClusterManager) constructor.newInstance();
+
+      clusterManager.initialize(conf);
+      setClusterManager(clusterManager);
```

Comment:
Why is `clusterManager` static ?

Reply:
The global cluster manager is static because it should be set only once. If erroneously someone is trying to set this, it should be ignored or throw exception

Reply:
OK. I checked the code. This seems wrong. Probably `final` is a better option but this is outside the scope of this PR.